### DISCUSSION TITLE
Selected state in a React component

### DISCUSSION THREAD

**Discussion by gchumillas** [tailwindlabs/tailwindcss#4290] [Help]
I'm creating a React component and I'd like to change the background color depending on whether it is selected or not. And it occurred to me that I could use a custom "state variant":

```jsx
<button selected={condition} className="bg-orange-900 selected:bg-green-900">
  Accept
</button>
```

Is that possible? Is there any other approach to solve this problem?

**Comment by dance2die** [comment]
You can use [focus](https://tailwindcss.com/docs/hover-focus-and-other-states#focus) variant to see if the button's "selected" (focused).

```html
<button class="btn m-10 focus:ring-2 active:bg-yellow-300 active:text-gray-500">Button!</button>
```

You can change background color when a button is "clicked" using [active](https://tailwindcss.com/docs/hover-focus-and-other-states#active) variant.

Check out the demo here: https://play.tailwindcss.com/iu4yKAIPiV

If you are making use of `condition` in your React code, you can conditionally render class,

```jsx
<button selected={condition} className={`bg-orange-900 ${condition ? "bg-green-900" : ""}`}>
  Accept
</button>
```

You can also simply the class manipulation using libraries, such as [classnames](https://www.npmjs.com/package/classnames) or [tailwind-classnames](https://www.npmjs.com/package/tailwindcss-classnames).

One thing to note is that when you conditionally update class, do not use partial names but use full class names  
because Tailwind CSS won't know if you use partial class names thus won't purge unused CSS.

### bad! :x: 

```jsx
<button selected={condition} className={`bg-green-${condition ? "500" : "900"}`}>
  Accept
</button>
```

### Good :heavy_check_mark: 
```jsx
<button selected={condition} className={`${condition ? "bg-green-500" : "bg-green-900"}`}>
  Accept
</button>
```
